File: AcmeDirectorySearch/Core/Models/Address.cs
namespace AcmeDirectorySearch.Core.Models
{
    using AcmeDirectorySearch.Core.Validations;
    using System.ComponentModel.DataAnnotations;

    /// <summary>
    /// Address object that reprsents a User object.  This model contains all of the
    /// business logic pertaining to Address
    /// </summary>
    public class Address : ValidationBase
    {
        [Required]
        [MinLength(5)]
        [MaxLength(100)]
        public string Street1 { get; set; }

        [MinLength(5)]
        [MaxLength(100)]
        public string Street2 { get; set; }

        [Required]
        [MinLength(5)]
        [MaxLength(50)]
        public string City { get; set; }

        [Required]
        [RegularExpression(@"^[A-Za-z]{2}$", ErrorMessage = "Invalid 'State' format, provide 2 letter abbreviation for State")]
        public string State { get; set; }

        [Required]
        [RegularExpression(@"^\d{5}$", ErrorMessage = "Invalid 'Zip' format, ##### required")]
        public string Zip { get; set; }
    }
}

File: AcmeUX/src/app/models/address.ts

/**
 * Interface and Model definitions of an Address
 *
 */
export interface IAddress {
  street1: string;
  street2: string | undefined;
  city: string;
  state: string;
  zip: string;
}

export class Address implements IAddress {

  street1: string;
  street2: string | undefined;
  city: string;
  state: string;
  zip: string;

  constructor(street1: string, street2: string|undefined, city: string, state: string, zip: string) {

    this.street1 = street1;
    this.street2 = street2;
    this.city = city;
    this.street2 = state;
    this.zip = zip;

  }

}
